apps/common-app/src/examples/RuntimeTests/ReJest/TestRunner/TestRunner.ts

```diff
@@ -1,30 +1,28 @@
-import type { Component, MutableRefObject, ReactElement } from 'react';
+import type { MutableRefObject, Component, ReactElement } from 'react';
 import { useRef } from 'react';
-import type { BuildFunction, TestCase, TestConfiguration, TestSuite, TestValue } from '../types';
-import { DescribeDecorator, TestDecorator } from '../types';
+import { assertExecutionManager } from './Asserts';
+import { RenderLock, SyncUIRunner } from '../utils/SyncUIRunner';
+import type { TestCase, TestConfiguration, TestSuite, TestValue } from '../types';
 import { TestComponent } from '../TestComponent';
-import { applyMarkdown, formatTestName } from '../utils/stringFormatUtils';
 import { Matchers } from '../matchers/Matchers';
-import { assertTestCase, assertTestSuite } from './Asserts';
-import { RenderLock } from '../utils/SyncUIRunner';
 import { ValueRegistry } from './ValueRegistry';
 import { TestSummaryLogger } from './TestSummaryLogger';
 import { WindowDimensionsMocker } from './WindowDimensionsMocker';
 import { AnimationUpdatesRecorder } from './AnimationUpdatesRecorder';
+import { TestSuiteExecutionManager, TestSuitePreExecutionManager } from './TestSuiteManager';
```

Comment:
Splitting this logic into two classes might be more confusing than keeping it within one class, especially since TestSuiteExecutionManager feels more like a struct than a class. 🤔